src/shadowbox/server/shared_metrics.ts

```diff
@@ -84,18 +86,48 @@ export class PrometheusUsageMetrics implements UsageMetrics {
 
   async getReportedUsage(): Promise<ReportedUsage[]> {
     const timeDeltaSecs = Math.round((Date.now() - this.resetTimeMs) / 1000);
-    // We measure the traffic to and from the target, since that's what we are protecting.
-    const result = await this.prometheusClient.query(
-      `sum(increase(shadowsocks_data_bytes_per_location{dir=~"p>t|p<t"}[${timeDeltaSecs}s])) by (location, asn)`
-    );
-    const usage = [] as ReportedUsage[];
-    for (const entry of result.result) {
-      const country = entry.metric['location'] || '';
-      const asn = entry.metric['asn'] ? Number(entry.metric['asn']) : undefined;
-      const inboundBytes = Math.round(parseFloat(entry.value[1]));
-      usage.push({country, inboundBytes, asn});
+    // Return both data bytes and tunnel time information with a single
+    // Prometheus query, by using a custom "metric_type" label.
+    const queryResponse = await this.prometheusClient.query(`
```

Comment:
I think this is an ok solution, though I don't consider good practice to mix data of different types. Here we are creating a single time series with different units and meaning. And later split them again. I'd rather keep them as separate time series, as it's a lot clearer.
I also wonder if this approach affects performance somehow.

Reply:
@fortuna Ok let me try something else. How about this?